core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -33,11 +34,15 @@ import kafka.security.auth.{Authorizer, ClusterAction, Group, Create, Describe,
 import kafka.utils.{Logging, SystemTime, ZKGroupTopicDirs, ZkUtils}
 import org.apache.kafka.common.metrics.Metrics
 import org.apache.kafka.common.protocol.{ApiKeys, Errors, SecurityProtocol}
-import org.apache.kafka.common.requests.{GroupCoordinatorRequest, GroupCoordinatorResponse, ListGroupsResponse, DescribeGroupsRequest, DescribeGroupsResponse, HeartbeatRequest, HeartbeatResponse, JoinGroupRequest, JoinGroupResponse, LeaveGroupRequest, LeaveGroupResponse, ResponseHeader, ResponseSend, SyncGroupRequest, SyncGroupResponse}
+import org.apache.kafka.common.requests.{ListOffsetRequest, ListOffsetResponse, GroupCoordinatorRequest, GroupCoordinatorResponse, ListGroupsResponse,
+DescribeGroupsRequest, DescribeGroupsResponse, HeartbeatRequest, HeartbeatResponse, JoinGroupRequest, JoinGroupResponse,
+LeaveGroupRequest, LeaveGroupResponse, ResponseHeader, ResponseSend, SyncGroupRequest, SyncGroupResponse}
 import org.apache.kafka.common.utils.Utils
-import org.apache.kafka.common.Node
+import org.apache.kafka.common.{TopicPartition, Node}
 
 import scala.collection._
+import scala.collection.JavaConverters._
```

Comment:
I moved `scala.collection.JavaConverters._` to the top since there doesn't appear to be any value to having it scoped and it only creates duplicate code.

I am happy to put `java.lang.Long` if you don't like the `JLong` rename. I just find the shorthand convenient when working with Java objects (JLong. JInt, JMap, etc).

I prefer to use JavaConverters over JavaConversions for a few reasons. And I think we should try to use JavaConverters exclusively:
- With JavaConverters it is more clear what is happening in your code
   - Debugging implicit conversions can be a pain
- JavaConverters is more type safe
   - ex: JavaConversions won't issue type errors when looking up values of the wrong type in Maps
- JavaConverters is more performant
   - There is no caching of implicit conversion, i.e. if the same conversion is applied twice inside a method, the conversion code is called twice.
   - The requirement to explicitly convert allows you to make sure you are only doing it once
- JavaConverters was added to Scala after JavaConversions for all the reasons mentioned above

We could also use https://github.com/scalaj/scalaj-collection which is an enhanced version of JavaConverters where the explicit/painful conversion of the inner java primitives is handled for you. Then I wouldn't need to map to java.util.Long. It's a small library and convenient, but not sure we really need it.

If you search around a bit you can find many more people recommending the use of JavaConverters. There is a more formal discussion explanation [here](http://freecontent.manning.com/wp-content/uploads/scala-java-interaction-be-wary-of-implicit-conversions.pdf) too.